app/src/main/java/com/nextlevelstudy/ui/common/NavigationController.java

```diff
@@ -0,0 +1,68 @@
+package com.nextlevelstudy.ui.common;
+
+import android.os.Build;
+import android.transition.Fade;
+import android.util.Log;
+import android.view.View;
+
+import androidx.core.view.ViewCompat;
+import androidx.fragment.app.FragmentManager;
+
+import com.nextlevelstudy.R;
+
+import com.nextlevelstudy.models.University;
+import com.nextlevelstudy.ui.MainActivity;
+import com.nextlevelstudy.ui.login.LoginFragment;
+import com.nextlevelstudy.ui.university.UniversityDetailFragment;
+import com.nextlevelstudy.ui.university.UniversityListFragment;
+
+import javax.inject.Inject;
+
+/**
+ * A utility class that handles navigation in {@link MainActivity}.
+ */
+
+public class NavigationController {
```

Comment:
Hmm.. since this is injected based on the Activity, I was hoping to see a scope qualifier being used.